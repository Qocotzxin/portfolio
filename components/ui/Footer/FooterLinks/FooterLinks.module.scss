.FooterLinks {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: var(--ce-space-4);

  & > * {
    margin-right: var(--ce-space-4);
  }

  &__icon {
    width: 24px;
    height: 24px;
    transition: color 0.2s;
    color: var(--ce-body-text);

    &:hover {
      color: var(--ce-color-blue-400);
    }
  }
}
