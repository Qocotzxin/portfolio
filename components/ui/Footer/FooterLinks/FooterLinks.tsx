import { HStack, Link } from "@chakra-ui/react";
import { useHygraphContext } from "@contexts/hygraphContext";
import { ICON_MAP } from "@models/hygraph";
import { FC } from "react";

const FooterLinks: FC = () => {
  const { footerLinks } = useHygraphContext();

  return (
    <HStack spacing={4} mr="4">
      {footerLinks?.map((link) => {
        const LinkIcon = ICON_MAP[link.icon];
        return (
          Boolean(LinkIcon) &&
          link.isActive && (
            <Link
              key={link.url}
              display="flex"
              href={link.url}
              isExternal
              aria-label={link.ariaLabel}
            >
              <LinkIcon _hover={{ color: "blue.400" }} />
            </Link>
          )
        );
      })}
    </HStack>
  );
};

export default FooterLinks;
