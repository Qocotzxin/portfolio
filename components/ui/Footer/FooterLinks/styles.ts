import styles from "./FooterLinks.module.scss";

export const Classes = {
  base: styles.FooterLinks,
  icon: styles.FooterLinks__icon,
};
