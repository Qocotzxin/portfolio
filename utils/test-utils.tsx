import { ChakraProvider } from "@chakra-ui/react";
import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import theme from "@theme";
import { HygraphProvider } from "contexts/hygraphContext";
import { Components, HygraphData } from "@models/hygraph";

export const hygraphMockData: Partial<HygraphData> = {
  skipLinks: [
    {
      text: "Skip to content",
    },
  ],
  ariaLabels: [
    {
      id: "1",
      metadata: "dark",
      component: Components.ColorModeSwitcher,
      content: "Switch to light mode",
    },
    {
      id: "2",
      metadata: "light",
      component: Components.ColorModeSwitcher,
      content: "Switch to dark mode",
    },
  ],
  navLinks: [
    {
      url: "test",
      order: 1,
      text: "Test",
    },
  ],
  languages: [
    { code: "en", displayName: "english", isActive: true },
    { code: "es", displayName: "spanish", isActive: true },
    { code: "nl", displayName: "dutch", isActive: false },
  ],
  footerLinks: [
    {
      url: "https://github.com",
      isActive: true,
      icon: "github",
      ariaLabel: "Visit Github",
    },
    {
      url: "https://test.com",
      isActive: false,
      icon: "github",
      ariaLabel: "Visit test",
    },
  ],
};

interface AllTheProvidersProps {
  hygraphData?: Partial<HygraphData>;
}

export const AllTheProviders: FC<PropsWithChildren & AllTheProvidersProps> = ({
  children,
  hygraphData = hygraphMockData,
}) => {
  return (
    <ChakraProvider theme={theme}>
      <HygraphProvider hygraphData={hygraphData}>{children}</HygraphProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  wrapperProps?: AllTheProvidersProps
) =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllTheProviders {...wrapperProps}>{children}</AllTheProviders>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { customRender as render };
