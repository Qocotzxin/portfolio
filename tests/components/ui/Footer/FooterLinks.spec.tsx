import { render, screen } from "@utils/test-utils";
import FooterLinks from "@ui/Footer/FooterLinks";

describe("FooterLinks", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<FooterLinks />);
    expect(container).toMatchSnapshot();
  });

  it("Should render active link icons with the information provided by Hygraph.", () => {
    render(<FooterLinks />);

    expect(screen.getByLabelText("Visit Github")).toBeInTheDocument();
    expect(screen.queryByLabelText("Visit test")).not.toBeInTheDocument();
  });
});
