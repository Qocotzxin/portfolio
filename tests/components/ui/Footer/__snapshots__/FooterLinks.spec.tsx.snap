// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLinks Should match snapshot. 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: var(--chakra-space-4);
}

.emotion-0>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: var(--chakra-space-4);
  margin-inline-start: var(--chakra-space-4);
}

.emotion-1 {
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-fast);
  transition-timing-function: var(--chakra-transition-easing-ease-out);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  color: inherit;
  font-family: var(--chakra-fonts-mono);
  font-size: var(--chakra-fontSizes-xs);
  font-weight: var(--chakra-fontWeights-bold);
  -webkit-transition: text-shadow 0.2s linear,color 0.2s linear;
  transition: text-shadow 0.2s linear,color 0.2s linear;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1:hover,
.emotion-1[data-hover] {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1:focus-visible,
.emotion-1[data-focus-visible] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-2 {
  width: 24px;
  height: 24px;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: currentColor;
}

.emotion-2:hover,
.emotion-2[data-hover] {
  color: var(--chakra-colors-blue-400);
}

<div>
  <div
    class="chakra-stack emotion-0"
  >
    <a
      aria-label="Visit Github"
      class="chakra-link emotion-1"
      href="https://github.com"
      rel="noopener"
      target="_blank"
    >
      <svg
        class="chakra-icon emotion-2"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
        />
      </svg>
    </a>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
